python/src/equistore/operations/reduce_over_samples.py

```diff
@@ -28,11 +28,43 @@ def _reduce_over_samples_block(
         assert sample in block_samples.names
 
     assert reduction in ["sum", "mean", "variance", "std"]
-
     # get the indices of the selected sample
     sample_selected = [
         block_samples.names.index(sample) for sample in remaining_samples
     ]
+
+    # checks if it is a zero sample TensorBlock
+    if block.samples.shape[0] == 0:
+        # Here is different from the general case where we use Labels.single() if
+        # if len(remaining_samples) == 0
+        # Labels.single() cannot be used because Labels.single() has not
+        # an np.empty() array as values but has one values, it has dimension (1,...)
+        # we want (0,...).
+        # here if len(remaining_samples) == 0 -> Labels([], shape=(0, 0), dtype=int32)
+
+        samples_label = Labels(
+            remaining_samples,
+            np.empty((0, len(remaining_samples)), dtype=np.int32),
```

Comment:
I think zeros is good enough here.

```suggestion
            np.zeros((0, len(remaining_samples)), dtype=np.int32),
```